pkg/build/api/types.go

```diff
@@ -111,6 +111,12 @@ type BuildSource struct {
 	// This allows to have buildable sources in directory other than root of
 	// repository.
 	ContextDir string `json:"contextDir,omitempty"`
+
+	// SourceSecretName is the name of a Secret that would be used for setting
+	// up the authentication for cloning private repository.
+	// The secret contains private ssh key, whereas the public key is associated
+	// with your SCM account.
+	SourceSecretName string `json:"sourceSecretName,omitempty"`
```

Comment:
Since @bparees insistent on having some description of the contents of the secret, I've reworked that sentence.